Testscripts/Windows/NESTED-KVM-STORAGE-PERF.ps1

```diff
@@ -35,132 +33,96 @@ function Start-TestExecution ($ip, $port)
 	$cmd = "/home/$user/${testScript} > /home/$user/TestExecutionConsole.log"
 	$testJob = Run-LinuxCmd -username $user -password $password -ip $ip -port $port -command $cmd -runAsSudo -RunInBackground
 
-	while ((Get-Job -Id $testJob).State -eq "Running" )
-	{
+	while ((Get-Job -Id $testJob).State -eq "Running") {
 		$currentStatus = Run-LinuxCmd -username $user -password $password -ip $ip -port $port -command "cat /home/$user/state.txt"
 		Write-LogInfo "Current Test Status : $currentStatus"
 		Wait-Time -seconds 20
 	}
 }
 
-function Send-ResultToDatabase ($GlobalConfig, $logDir)
-{
-	Write-LogInfo "Uploading the test results.."
-	$dataSource = $GlobalConfig.Global.$TestPlatform.ResultsDatabase.server
-	$DBuser = $GlobalConfig.Global.$TestPlatform.ResultsDatabase.user
-	$DBpassword = $GlobalConfig.Global.$TestPlatform.ResultsDatabase.password
-	$database = $GlobalConfig.Global.$TestPlatform.ResultsDatabase.dbname
-	$dataTableName = $GlobalConfig.Global.$TestPlatform.ResultsDatabase.dbtable
-	$TestCaseName = $GlobalConfig.Global.$TestPlatform.ResultsDatabase.testTag
-	if ($dataSource -And $DBuser -And $DBpassword -And $database -And $dataTableName)
-	{
-		$fioDataCsv = Import-Csv -Path $LogDir\fioData.csv
-		$HostType = $TestPlatform
-		if ($TestPlatform -eq "hyperV")
-		{
-			$HostBy = $TestLocation
-			$HyperVMappedSizes = [xml](Get-Content .\XML\AzureVMSizeToHyperVMapping.xml)
-			$L1GuestCpuNum = $HyperVMappedSizes.HyperV.$HyperVInstanceSize.NumberOfCores
-			$L1GuestMemMB = [int]($HyperVMappedSizes.HyperV.$HyperVInstanceSize.MemoryInMB)
-			$L1GuestSize = "$($L1GuestCpuNum)Cores $($L1GuestMemMB/1024)G"
+function Send-ResultToDatabase ($currentTestResult) {
+	$fioDataCsv = Import-Csv -Path $LogDir\fioData.csv
+
+	if ($TestPlatform -eq "hyperV") {
+		$HostBy = $TestLocation
+		$HyperVMappedSizes = [xml](Get-Content .\XML\AzureVMSizeToHyperVMapping.xml)
+		$L1GuestCpuNum = $HyperVMappedSizes.HyperV.$HyperVInstanceSize.NumberOfCores
+		$L1GuestMemMB = [int]($HyperVMappedSizes.HyperV.$HyperVInstanceSize.MemoryInMB)
+		$L1GuestSize = "$($L1GuestCpuNum)Cores $($L1GuestMemMB/1024)G"
+	} else {
+		$HostBy	= ($global:TestLocation).Replace('"','')
+		$L1GuestSize = $AllVMData.InstanceSize
+	}
+	$count = 0
+
+	# TODO: Change to get the disk size from either host side or guest side
+	<#
```

Comment:
could you fix this? to get the real disk size

Reply:
Can we retrieve it from VM configuration? 
```      
                <DataDisk>
                    <LUN>0</LUN>
                    <DiskSizeInGB>513</DiskSizeInGB>
                    <HostCaching>None</HostCaching>
                </DataDisk>
```

Reply:
it's the previous approach, the problem is that the case script should not know the VM setup type, as it's managed by the framework to deploy VMs

Reply:
ok.